### DISCUSSION TITLE
"Generate Oauth Token" Fails

### DISCUSSION THREAD

**Discussion by stellarhopper** [llamafilm/tesla-http-proxy-addon#46] [Q&A]
Hello,

I'm setting this up, and I got everything to the point of going to the HA companion app (I'm on android), and clicking Generate Oauth Token.

It opens the web browser page, I can login, but the redirect URL I get after that (https://example.com/callback?...) gives me a `404 not found`.

Any suggestions on what I may have missed?

I already have SSL + letsencrypt setup for external access, so I didn't create a new tesla.example.com based on another discussion here. I set up the tesla developer portal with my existing 'example.com' and it was happy.

Edit:
Ah I see an error in the add-on logs saying public key not found  at the .well-known location. Indeed if I navigate to that URL, I get a 404. Did I miss / screw up a step?

**Comment by llamafilm** [comment]
Are you running on Home Assistant OS?
Have you tried restarting nginx?
Does that public key file exist and have valid contents?

**Comment by stellarhopper** [comment]
@llamafilm correct, HAOS on the Home assistant Yellow board
I tried restarting all of HA, still nothing.
Where do I like for the file? I don't see anything in /homeassistant/www and searching for a '.well-known' also returns nothing.

**Reply by llamafilm** [reply]
Each addon runs in a separate container, so restarting HA will not restart Nginx. That’s what I said you need to restart Nginx so that it loads the new config.

The file you found is in the correct path. Does it have the correct contents?

**Comment by stellarhopper** [comment]
I noticed in my first attempt I misread the instructions and neglected to start the addon after just entering the domain.
I uninstalled and reinstalled it, and did that this time around, got log messages saying the SSL cert, config, and public key were generated.

I can find the config in /share/nginx_proxy/nginx_tesla.conf

However /.well-known doesn't exist. Not sure why it didn't get created - perhaps I can create the directories it expects, and paste in the public key (from the logs) manually to see if it gets me further.

**Comment by stellarhopper** [comment]
Creating a /.well-known/... etc path manually didn't work, but I did find the key in /share/tesla/com.tesla.3p.public-key.pem

**Comment by baylanger** [comment]
You don’t see to create /.well-known , the config defines the location (path) within the docker container.

https://github.com/llamafilm/tesla-http-proxy-addon/blob/b4fe19013579b4e4d5e74e446ef7d6dd2e76747a/tesla_http_proxy/app/nginx_tesla.conf#L21

**Comment by stellarhopper** [comment]
Ah ok - so I rebooted everything, and redid the setup. After adding the
client id/secred in the addon config, I still get an error sawying the
public key URL returned a 404. However navigating to the URL (i.e. the
example.com/.well-known/… one) does work and downloads the key.

Navigating to the /callback URL after the Oauth step from Web UI retuens a
404.

**Comment by stellarhopper** [comment]
BTW when I restart the nginx proxy addon, I see a warning:
`nginx: [warn] conflicting server name "<example.com>" on 0.0.0.0:443, ignored`
Is this relevant?

**Reply by llamafilm** [reply]
Yes that sounds like a problem

**Reply by Briochorama** [reply]
I'm in the exact same situation. Have you found a way to move forward ?

**Reply by Briochorama** [reply]
About the `nginx: [warn] conflicting server name` error. It looks like the addon is loading two config files or that the definition of `servers` is redundant, however I can't understand why. The config appears to load the correct file :

```
domain: mydomain.duckdns.org
hsts: max-age=31536000; includeSubDomains
certfile: fullchain.pem
keyfile: privkey.pem
cloudflare: false
customize:
  active: true
  default: nginx_proxy_default*.conf
  servers: nginx_proxy/nginx_tesla.conf
```

But there's no duplicate definition in its content 
![image](https://github.com/llamafilm/tesla-http-proxy-addon/assets/12542442/0cd0517b-149e-4f21-a94f-4a78f1448b43)

I _think_ there's another config file somewhere else that is somehow loadeed as well, but I can't figure out where.

**Reply by llamafilm** [reply]
Hiding your domain name doesn't maker you any safer, it just makes it difficult for anyone to help you.

**Reply by Briochorama** [reply]
At the moment, the url redirects to the login page of my home assistant instance (that I can also access via my nabu casa url). Would it be really helpful and safe to share it ?

**Comment by stellarhopper** [comment]
Nope, I gave up for now - my car hasn't been switched to the new API (old
one still works), and another thread suggested we need to wait until the
switch is flipped before we can use the Fleet API + http proxy? Anyway I'll
revisit it when I need to, but otherwise I didn't make any progress.

**Reply by llamafilm** [reply]
There is no need to wait.  Fleet API is fully functional, although it is complicated to setup and that make it unattainable for some people.

**Reply by Briochorama** [reply]
Thanks. It seems that Tesla just announced the retirement of the old API. Like gone for good this time. Things is, the documentation for the addon seems to be ok, it's the prerequisites I'm having trouble with (like how to setup nginx + duckdns/ nabu casa).

Hopefully we'll get a solution soon. Then I can write down some kind of procedure/tutorial for people who are struggling like I am.

**Comment by stellarhopper** [comment]
The setup was a bit unclear to me around what's needed for the https / cert
side of things. I already have a https / letsencrypt setup using the Nginx
reverse proxy, that's how I access HA. Do I still neea a new tesla.domain?
Or can I keep using my existing top level domain for the proxy setup?
